src/plugins/data/server/index_patterns/routes.ts

```diff
@@ -70,7 +70,21 @@ export function registerRoutes(http: HttpServiceSetup) {
           },
         });
       } catch (error) {
-        return response.notFound();
+        if (
+          !!error.isBoom &&
+          !!error?.output?.payload &&
```

Comment:
nit: is line 75 necessary? seems it may be redundant with line 76

Reply:
Yes, it is necessary, one like checks that payload is object, another makes sure the object is not falsy&mdash;`null`.